src/iticket.cpp

```diff
@@ -653,8 +655,13 @@ doCommand( char *cmdToken[] ) {
         else {
             makeTicket( myTicket );
         }
-        std::string fullPath = makeFullPath( cmdToken[2] );
-        doTicketOp( "create", myTicket, cmdToken[1], fullPath.c_str(),
+
+        auto fullPath = canonical(cmdToken[2]);
+        if (!fullPath.has_value()) {
+            return -2;
```

Comment:
How did we arrive at `-2`?

Reply:
My guess is because there's a -1 above this

Reply:
I had a similar thought, but perhaps @Nishant2022 had a different reason.

Reply:
At the top of the current function there is a comment saying:

```c++
/* handle a command,
   return code is 0 if the command was (at least partially) valid,
   -1 for quitting,
   -2 for if invalid
   -3 if empty.
 */
``` 

If the canonical function does not return a value, then the provided path is invalid and the return value should reflect that.

Reply:
Sounds good to me.

Reply:
Please mark this as resolved.